src/natcap/invest/forest_carbon_edge_effect.py

```diff
@@ -498,41 +497,23 @@ def _aggregate_carbon_map(
 
     if os.path.exists(target_aggregate_vector_path):
         os.remove(target_aggregate_vector_path)
-    target_aggregate_vector = driver.CreateCopy(
-        target_aggregate_vector_path, aoi_vector)
+    driver.CreateCopy(target_aggregate_vector_path, aoi_vector)
     aoi_vector = None
-    target_aggregate_layer = target_aggregate_vector.GetLayer()
 
-    # make an identifying id per polygon that can be used for aggregation
-    while True:
-        serviceshed_defn = target_aggregate_layer.GetLayerDefn()
-        poly_id_field = str(uuid.uuid4())[-8:]
-        if serviceshed_defn.GetFieldIndex(poly_id_field) == -1:
-            break
-    layer_id_field = ogr.FieldDefn(poly_id_field, ogr.OFTInteger)
-    target_aggregate_layer.CreateField(layer_id_field)
-    target_aggregate_layer.StartTransaction()
-    for poly_index, poly_feat in enumerate(target_aggregate_layer):
-        poly_feat.SetField(poly_id_field, poly_index)
-        target_aggregate_layer.SetFeature(poly_feat)
-    target_aggregate_layer.CommitTransaction()
-    target_aggregate_layer.SyncToDisk()
-
-    # aggregate carbon stocks by the new ID field
```

Comment:
`zonal_statistics` aggregates by the FID, this random ID field wasn't being used.